### ISSUE TITLE
Xcode apps no longer compile

### ISSUE THREAD

**Issue by MatthiasWM** [fltk/fltk#1103]
**Describe the bug**
No apps compile anymore using the CMake setup on macOS due to ScreenCapture library no being linked anymore.

**To Reproduce**
- Update Xcode to 16.0, 
- delete the build folder
- rebuild the cmake build folder using `cmake -B build/Xcode -G Xcode .`
- launch Xcode, compile any test app, and you get 4 linke errors similar to: `Undefined symbol: _OBJC_CLASS_$_SCContentFilter`

**FLTK Version**
 - Version: 1.4.0 master
 - If from Git, branch: master

**FLTK Configure / Build Options**
Please add the complete configure and build command lines if you built FLTK yourself.
Please include all configure or CMake options and add all CMake options you changed
by using CMake GUI tools (if any).
 - `cmake -B build/Xcode -G Xcode .`

**Operating System / Platform:**
- macOS 14.6.1 (23G93)
- Xcode Version 16.0 (16A242d)
- cmake version 3.30.2
- uname -r  = 23.6.0
- CMake ${CURRENT_OSX_VERSION} 14.6.1

**Additional context**
In CMake/Ssetup.CMake, line 130 reads `if(${CMAKE_SYSTEM_VERSION} VERSION_GREATER_EQUAL 24.0)` but the variable is set to 23.6.0 on my system after the upgrade to Xcode 16.0. I upgraded all my Brew stuff and got CMake 3.30.5, but CMAKE_SYSTEM_VERSION stays at 23.6.0. As a result, `-weak_framework ScreenCaptureKit` is not added to FLTK_COCOA_FRAMEWORKS.

My macOS docs say, ScreenCaptureKit was introduced with macOS 12.3.

Would it make more sense to query CMAKE_OSX_DEPLOYMENT_TARGET in the CMake file, which is 14.6 on my machine, for example?

**Comment by Albrecht-S** [comment]
Note: FLTK 1.4.0 RC3 is delayed until this bug has been fixed. Hopefully soon ;-)

**Comment by MatthiasWM** [comment]
@ManoloFLTK 

Puttingg together more information. So CMAKE_OSX_DEPLOYMENT_TARGET changes MAC_OS_X_VERSION_MIN_REQUIRED, but the requirement to link the library depends on MAC_OS_X_VERSION_MAX_ALLOWED. I am reading up on those macros now, hoping I can give a solutions quickly.

**Comment by MatthiasWM** [comment]
Ok, here is a suggestion. Please @ManoloFLTK and @Albrecht-S , let me know what you think:

ScreenCaptureKit is only compiled in if MAC_OS_X_VERSION_MAX_ALLOWED is 15.0 or higher. If that is the case, we must also link the framework, but there is no way to query MAC_OS_X_VERSION_MAX_ALLOWED from CMake.

But we can still get this number by writing a tiny program that outputs MAC_OS_X_VERSION_MAX_ALLOWED, using the CMake "try_run" call. That way we can synchronize the compiling and linking of features.

Does that make sense?

**Comment by ManoloFLTK** [comment]
I see Matthias' proposal after having undone commit 6f6a375 into commit 37144c0.

This proposal seems very good, I will look at it now.

FTR: use of the ScreenCaptureKit framework began because the API used to capture window titlebars was not only deprecated by even obsolete in SDK 15.0 beta. Using that obsolete API produced big error messages when running macOS 15.0. The adequate API  to capture screen data in that SDK is inside ScreenCaptureKit. This is complicated by a bizarre change in SDK 15.0 where MAC_OS_X_VERSION_MAX_ALLOWED is no longer defined by the SDK file AvailabilityMacros.h to 150000 (it stays at 140000). I found recently that SDK 15.0 instead defines __MAC_OS_X_VERSION_MAX_ALLOWED (that is done inside AvailibilityInternal.h) to the correct 150000 value. That was the meaning of commit 6f6a375.

But commit 6f6a375 revealed that the current approach FLTK follows to support frameworks used in recent macOS versions while remaining compatible with more ancient versions is incomplete in CMake files. Matthias post above describes exactly the nature of the problem.

**Comment by ManoloFLTK** [comment]
Regarding Matthias' try_run-based proposal

- It works beautifully. That's great.

- The handling of the UniformTypeIdentifiers framework by CMake should be changed also
```
      if(${CMAKE_SYSTEM_VERSION} VERSION_GREATER_EQUAL 20.0) # a.k.a. macOS version ≥ 11.0
        list(APPEND FLTK_COCOA_FRAMEWORKS "-weak_framework UniformTypeIdentifiers")
      endif()
```
should become
```
      if(${MAC_OS_X_VERSION_MAX_ALLOWED} VERSION_GREATER_EQUAL 11.0)
        list(APPEND FLTK_COCOA_FRAMEWORKS "-weak_framework UniformTypeIdentifiers")
      endif()
```


- I see this in the documentation of CMAKE_SYSTEM_VERSION
> In the case of a host build then CMAKE_SYSTEM_VERSION may be set explicitly when first configuring a new build tree in order to enable targeting the build for a different version of the host operating system than is actually running on the host. 

Does that suggest another way to proceed?

- I want to test in detail what happens under macOS 15 (later today).

**Comment by ManoloFLTK** [comment]
Regarding Matthias' try_run-based proposal - cont'd:

the configure-based build procedure needs adaptation too.

**Comment by ManoloFLTK** [comment]
Here are my findings when booting into macOS 15.1:
```
============== CMake-based build ==================

Matthias' code builds and runs well under macOS 15. ScreenCaptureKit is weak linked and used at run-time.
Error at build time if attempting to compile CGWindowListCreateImageFromArray() with SDK 15.

Matthias' code built under macOS 14 with SDK 15 and run under macOS 15 is OK: 
ScreenCaptureKit is used at run-time.

Matthias' code includes this : 
#ifdef   __MAC_OS_X_VERSION_MAX_ALLOWED
#  undef   MAC_OS_X_VERSION_MAX_ALLOWED
#  define  MAC_OS_X_VERSION_MAX_ALLOWED  __MAC_OS_X_VERSION_MAX_ALLOWED
#endif

This redefinition of MAC_OS_X_VERSION_MAX_ALLOWED is necessary to access SDK 15
Without it
MAC_OS_X_VERSION_MAX_ALLOWED=140000 __MAC_OS_X_VERSION_MAX_ALLOWED=150000
even on macOS 15.1, same thing on macOS 14.x


FLTK without the patch that redefines MAC_OS_X_VERSION_MAX_ALLOWED from __MAC_OS_X_VERSION_MAX_ALLOWED
builds against SDK 15 and runs OK under macOS 14 and macOS 15, but does not use ScreenCaptureKit
at run-time because it gets MAC_OS_X_VERSION_MAX_ALLOWED=140000 which excludes ScreenCaptureKit
from compilation.

============== configure-based build ==================
FLTK as in github builds and runs OK under macOS 15. ScreenCaptureKit is used at run-time.

Matthias' code builds and runs OK under macOS 15. ScreenCaptureKit is used at run-time.
But it fails at build-time when built under macOS 14 if the last SDK (15) was loaded.

These are the values seen when using configure/make
MAC_OS_X_VERSION_MAX_ALLOWED=150000 __MAC_OS_X_VERSION_MAX_ALLOWED=150100
It seems that, when no explicit SDK option is given in the compile/link command,
MAC_OS_X_VERSION_MAX_ALLOWED equals 150000 as expected for SDK 15.
```

**Comment by ManoloFLTK** [comment]
Commit afe87de does for configure-based builds what Matthias' proposed code does for CMake-based builds: use a test program to detect whether __MAC_OS_X_VERSION_MAX_ALLOWED is high enough to require an additional framework to be weakly linked in.

With this, I believe Matthias' proposal to be the way to go, after extending it to also handle the UniformTypeIdentifiers framework.

**Comment by Albrecht-S** [comment]
FWIW: I tested with the latest commit (afe87de0131903e5e4c0d6bbfd553560fd7f89e9) and could build on my macOS (M1) system after I updated Xcode.
- macOS Sequioa 15.0.1
- SDK 15.1
- Xcode 16.1
- `xcode-select --install` issues: "Command line tools are already installed..."
- [x] build with Xcode: OK
- [x] build with configure/make: OK
- [x] build with Ninja: OK

I also tried setting different `CMAKE_OSX_DEPLOYMENT_TARGET` values like `CMAKE_OSX_DEPLOYMENT_TARGET:STRING=12.0` or even an empty value [1].

However, testing with `CMAKE_OSX_DEPLOYMENT_TARGET:STRING=15.0` fails whereas "14.0" works.

[1] If I remove `CMAKE_OSX_DEPLOYMENT_TARGET` from the cache CMake creates an empty value and the build with Ninja and Xcode both fail.

I hope this helps. I didn't try Matthias' proposal, I'm leaving this to you, @ManoloFLTK and @MatthiasWM.

**Comment by MatthiasWM** [comment]
Thanks guys! Very much appreciated! I just committed Albrechts suggestions and checked with my configuration with CMake Xcode, Ninja, and Makefiles, with the default deployment target and `-D CMAKE_OSX_DEPLOYMENT_TARGET=14.6.1` and all seems fine (FLUID compiles and runs). I have not checked autoconf/Make though.

I love Manolos `configure.ac` version which doesn't;t even need to run anything, just try to compile. Nice.

As for `__MAC_OS_X_VERSION_MAX_ALLOWED` with no underscores vs. two underscores, it seems that underscore version is in `Availability.h` and the non-underscore is in `AvailabilityMacros.h`, but the preferred method in the Apple code  saw is using the underscored one. Should we change that for the entirety of the code to avoid the cludge of redefining `MAC_OS_X_VERSION_MAX_ALLOWED`? Apple does not seem too diligent to verify that macro. Same for `...MIN_REQUIRED`.

Thanks of the UniformTypeIdentifiers changes. That make much sense!

One point of failure I found: If a user is cross-compiling, my little app can't run and the fallback is used. Is it better to use `try_compile` instead of `try_run`, just like in Manolos `configure.ac` version? Can `try_compile` maybe even return an error code that contains the value of the macro?

**Comment by Albrecht-S** [comment]
> As for __MAC_OS_X_VERSION_MAX_ALLOWED with no underscores vs. two underscores, it seems that underscore version is in Availability.h and the non-underscore is in AvailabilityMacros.h, but the preferred method in the Apple code saw is using the underscored one.

Is the "underscored one" also defined in older macOS / SDK versions ?

> One point of failure I found: If a user is cross-compiling, my little app can't run and the fallback is used. Is it better to use try_compile instead of try_run, just like in Manolos configure.ac version?

If `try_compile` is sufficient, then yes - for the reason you mentioned.

> Can try_compile maybe even return an error code that contains the value of the macro?

I don't think so, see docs: https://cmake.org/cmake/help/v3.15/command/try_compile.html

The only usable output variable seems to be
> OUTPUT_VARIABLE <var>
>    Store the output from the build process in the given variable.

... which wouldn't give us a useful values, would it?

Look for "Try Compiling Source Files" and note that I **_intentionally_** linked version 3.15 of the docs to be sure it works in our minimal CMake version (although code that runs only on macOS is maybe less restricted?).

**Comment by MatthiasWM** [comment]
See PR #1105

**Comment by MatthiasWM** [comment]
Improved version that is slower, but works with cross compiling.

**Comment by ManoloFLTK** [comment]
Regarding MAC_OS_X_VERSION_MAX_ALLOWED vs __MAC_OS_X_VERSION_MAX_ALLOWED
> the preferred method in the Apple code saw is using the underscored one.

@MatthiasWM Where did you see that?

I see inside file AvailabilityMacros.h of SDK 15.0
```
                 This header enables a developer to specify build time
                 constraints on what Mac OS X versions the resulting
                 application will be run.  There are two bounds a developer
                 can specify:
                 
                      MAC_OS_X_VERSION_MIN_REQUIRED
                      MAC_OS_X_VERSION_MAX_ALLOWED
```
In contrast, inside Availability.h which defines __MAC_OS_X_VERSION_MAX_ALLOWED :
```
These macros are for use in OS header files.
```

Here is what I get under macOS 15.1 Sequoia and Xcode 16.0 with the FLTK code at the top of master:
```
* with CMAKE_OSX_DEPLOYMENT_TARGET=15.0 or undefined
MAC_OS_X_VERSION_MAX_ALLOWED=150000 __MAC_OS_X_VERSION_MAX_ALLOWED=150000
and ScreenCaptureKit is used at run-time.

* with CMAKE_OSX_DEPLOYMENT_TARGET=14.0 or 11.0
MAC_OS_X_VERSION_MAX_ALLOWED=140000 __MAC_OS_X_VERSION_MAX_ALLOWED=150000
and ScreenCaptureKit is not used at run-time.
```

This is totally baffling to me. Anyway, we get apps running OK under macOS Sequoia in all combinations, and they can be built to be compatible with a range of macOS versions. Thus we are good.

> Should we change that (MAC_OS_X_VERSION_MAX_ALLOWED vs __MAC_OS_X_VERSION_MAX_ALLOWED) for the entirety of the code to avoid the cludge of redefining MAC_OS_X_VERSION_MAX_ALLOWED? 

My preference at this point in time is not to change and keep MAC_OS_X_VERSION_MAX_ALLOWED as defined by AvailabilityMacros.h. I'd like to really understand the meaning of MAC_OS_X_VERSION_MAX_ALLOWED and __MAC_OS_X_VERSION_MAX_ALLOWED having sometimes distinct values before changing all the macOS FLTK code base. We see that we get working apps in all combinations of build parameters with the present setup.

I'd say we can close this issue and proceed with 1.4rc3.

**Comment by ManoloFLTK** [comment]
> Is the "underscored one" also defined in older macOS / SDK versions ?

@Albrecht-S I have all macOS SDKs since SDK 10.3. The second, underscored form appeared in SDK 10.5 which is quite old.

**Comment by Albrecht-S** [comment]
@ManoloFLTK wrote:
> My preference at this point in time is not to change and keep MAC_OS_X_VERSION_MAX_ALLOWED as defined by AvailabilityMacros.h. ...We see that we get working apps in all combinations of build parameters with the present setup.

Yes, fortunately we require and test for SDK 15.0 (not 16.0 yet). Since the two variables appear to be the same in 15.0 our tests succeed.

> I'd say we can close this issue and proceed with 1.4rc3.

Note that I pushed another commit (d5c0e215a46a25de58e6264280c74d428404075f). The changes are mostly (CMake-) technically regarding cache variables and better encapsulating by using a function instead of a macro.

@MatthiasWM I hope you don't mind that I changed all this. Some parts look a little more complicated, particularly the part that deals with using different cache variables.

Another change is to test for SDK 15.0 first and skip the test for 11.0 if 15.0 is available. ;-)

Please test and confirm that this commit is OK, then I'll start the release process for 1.4.0rc3.

**Comment by Albrecht-S** [comment]
@MatthiasWM You wrote:
> Improved version that is slower, but works with cross compiling.

Why do you think that a version that does **only build** a program is slower than one that **builds and runs** it? Or am I missing anything?

**Comment by Albrecht-S** [comment]
Sorry, I had to push yet another commit (87527017b26fadae156f99c7f356aab5a60c7c27). The previous `BINDIR` of the SDK test would have been outside the current build folder (this was a typo and not intended).

**Comment by Albrecht-S** [comment]
@ManoloFLTK Would you like to "optimize" `configure.ac` also by testing for 15.0 first and for 11.0 only if 15.0 is not available?

**Comment by MatthiasWM** [comment]
> @MatthiasWM You wrote:
> 
> > Improved version that is slower, but works with cross compiling.
> 
> Why do you think that a version that does **only build** a program is slower than one that **builds and runs** it? Or am I missing anything?

The old version compiles a single program that returns the value. With try_compile, I can reasonably only return true or false, so we have to check against every required version number (luckily only two at the moment). Two compiles are slower than one compile and one run.

**Comment by MatthiasWM** [comment]
PS: in your version, if the user is on the most modern system, the try_compile happens only once, of course.

**Comment by Albrecht-S** [comment]
@MatthiasWM OK, that (two builds vs one build and run) was what I was missing. Thanks.

**Comment by ManoloFLTK** [comment]
@Albrecht-S 
> Would you like to "optimize" configure.ac also by testing for 15.0 first and for 11.0 only if 15.0 is not available?

No, I won't do that because it would complicate the code for a minute benefit. Also, the configure-based approach is close to its end of life.

**Comment by ManoloFLTK** [comment]
Regarding the possibility of backporting framework-related stuff to 1.3:

Framework `UniformTypeIdentifiers` is used in 1.4 to remove a deprecation message. The deprecated macOS method is still used in 1.3. SInce 1.3 is not expected to live much on future macOS versions, that's OK.

1.3 does not capture window titlebars from the screen. Instead it draws titlebars to a non-display drawing surface. Therefore framework `ScreenCaptureKit` is of no use to 1.3. The 1.4 approach is more complicated but produces a titlebar with its correct rounded corners. That's not available in 1.3.

Overall, there's no need to backport any of this to 1.3.

**Comment by ManoloFLTK** [comment]
@Albrecht-S 
> Yes, fortunately we require and test for SDK 15.0 (not 16.0 yet). 

I don't understand your point. What is fortunate there? We test for SDK 15.0 because that's the first SDK where a certain API is obsolete (not only deprecated) and replaced by a new API from the `ScreenCaptureKit`framework. Presumably, that new API will live more than one year and will be available also in 16.0 next year and later.

> Since the two variables appear to be the same in 15.0 our tests succeed.

Here also I don't follow you. My post above shows that the two variables MAC_OS_X_VERSION_MAX_ALLOWED and __MAC_OS_X_VERSION_MAX_ALLOWED sometimes differ, sometimes have the same value, depending on the value of CMAKE_OSX_DEPLOYMENT_TARGET, all of this being done with a constant SDK 15.0. This, I find very odd because MAC_OS_X_VERSION_MAX_ALLOWED is supposed to be fixed by the version of the SDK in use. That's the way it has been until the appearance of SDK 15.0 and macOS 15 Sequoia.

**Comment by Albrecht-S** [comment]
@ManoloFLTK OK, I agree that it wouldn't be useful to change the configure-based stuff in 1.4.

> The deprecated macOS method is still used in 1.3. Since 1.3 is not expected to live much on future macOS versions, that's OK.

I hope you are right with this assumption. At least we'd have a porting target in 1.4.0 or later.

> ... there's no need to backport any of this to 1.3.

Great, that's what I wanted to hear. This should make it much easier to **finish** branch-1.3 soon.

**Comment by Albrecht-S** [comment]
I wrote:
> ... fortunately we require and test for SDK 15.0 (not 16.0 yet).

@ManoloFLTK replied:
> What is fortunate there? We test for SDK 15.0 because that's the first SDK where a certain API is obsolete (not only deprecated) ...

That's exactly what I meant: we are **lucky** that we only **need** to test for 15.0 and not 16.0. BTW, as it stands, our test program in CMake `try_compile()` tests for the "underscored" version of the macro which would likely work with 16.0 as well (or not, it doesn't matter). I'm not sure what the related test in the C++ or ObjectiveC code does.

Maybe I was not clear in my wording, but if you still don't get what I **meant**, just forget it. It's OK as it is now.

> My post above shows that the two variables MAC_OS_X_VERSION_MAX_ALLOWED and __MAC_OS_X_VERSION_MAX_ALLOWED sometimes differ, sometimes have the same value, depending on the value of CMAKE_OSX_DEPLOYMENT_TARGET, all of this being done with a constant SDK 15.0.

Yes, that's really weird, I agree.

May we assume that either
- this is a bug (oversight) in the SDK header files and will be fixed in a future release, or
- this is the **first** SDK where this is so, and only one of these macros will be maintained in the future?

Whatever the answer is, it doesn't matter now, let's see what happens in the future...

**Comment by Albrecht-S** [comment]
One last question: Are we OK with commit 87527017b26fadae156f99c7f356aab5a60c7c27? Is it OK to release 1.4.0rc3?

- [ ] Matt: yes, I tested it and it's OK
- [ ] Matt: NO (please add a comment)
- [ ] Manolo: yes, I tested it and it's OK
- [ ] Manolo: NO (please add a comment)

Please tick the check box above or post a comment.

I'm not sure if I can find the time to do it tomorrow, maybe in the afternoon or evening.

**Comment by MatthiasWM** [comment]
I tested my unchanged configuration and it works flawlessly with CMake and the current API (15.0.0) and also if I set -DCMAKE_OSX_DEPLOYMENT_TARGET=12.0 . I have not tested the autoconf/configure setup because I remember having issues with a faulty installation of limping on my side. Fingers crossed and thumbs squeezed for RC3!

**Comment by Albrecht-S** [comment]
@MatthiasWM Thanks for testing and feedback.

FWIW, I tested default `configure/make`, `CMake + Ninja` (default build), and `CMake + Ninja` (X11 aka XQuartz). Everything appeared to work for me. I don't remember if I tested different `CMAKE_OSX_DEPLOYMENT_TARGET` values after my latest commit (and if yes: which ones).

**Comment by ManoloFLTK** [comment]
@Albrecht-S I tested building 8752701 with CMake and configure both under macOS 14 and macOS 15. All work flawlessly.

**Comment by ManoloFLTK** [comment]
@MatthiasWM I'm sorry to insist, but I'm very interested in clarifying whether __MAC_OS_X_VERSION_MAX_ALLOWED is the preprocessor variable recommended by Apple. What makes you believe this?

**Comment by ManoloFLTK** [comment]
@Albrecht-S 
> we are lucky that we only need to test for 15.0 and not 16.0. 

What we do is really "test for 15.0 or more", because an API appears in a version and stays in higher versions, a few of them at least.

> Maybe I was not clear in my wording, but if you still don't get what I meant, just forget it. 

Yes, I still don't get what you mean. And yes, I will just forget it.

**Comment by MatthiasWM** [comment]
The 16.0 is the version of the Xcode app. 15.0.0 is the current macOS SDK. Both numbers are unrelated. Sorry if those informations caused confusion.

As for MAC_OS_X_VERSION_MAX_ALLOWED vs. __ MAC_OS_X_VERSION_MAX_ALLOWED, the value depends on `-mmacosx-version-min` wich in turn sets AFAIK `MAC_OS_X_VERSION_MIN_REQUIRED`.  The C++ version seems to have an impact as well, so C++20 requires at least a 13.3 deployment target according to some GitHub comment.

In any case, the question if we link a library or not should query the same macro that is used to determine if the source code that uses the library is linked in. So should we change the `macOSMaxAllowed.c` from underscored to non-underscored?

See https://github.com/fltk/fltk/blob/87527017b26fadae156f99c7f356aab5a60c7c27/CMake/macOSMaxAllowed.c#L3 vs https://github.com/fltk/fltk/blob/87527017b26fadae156f99c7f356aab5a60c7c27/src/Fl_cocoa.mm#L52

This is the non-underscore code in AvailabilityMacros.h:

```
#ifndef MAC_OS_X_VERSION_MAX_ALLOWED
    #if MAC_OS_X_VERSION_MIN_REQUIRED > MAC_OS_VERSION_14_0
        #define MAC_OS_X_VERSION_MAX_ALLOWED MAC_OS_X_VERSION_MIN_REQUIRED
    #else
        #define MAC_OS_X_VERSION_MAX_ALLOWED MAC_OS_VERSION_14_0
    #endif
#endif
```

The underscore code is in Availability.h:

```
#ifndef __MAC_OS_X_VERSION_MIN_REQUIRED
    #if defined(__has_builtin) && __has_builtin(__is_target_os)
        #if __is_target_os(macos)
            #define __MAC_OS_X_VERSION_MIN_REQUIRED __ENVIRONMENT_OS_VERSION_MIN_REQUIRED__
            #define __MAC_OS_X_VERSION_MAX_ALLOWED __MAC_15_0
        #endif
    #elif  __ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__ 
        #define __MAC_OS_X_VERSION_MIN_REQUIRED __ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__
        #define __MAC_OS_X_VERSION_MAX_ALLOWED __MAC_15_0
    #endif /*  __has_builtin(__is_target_os) && __is_target_os(macos) */
#endif /* __MAC_OS_X_VERSION_MIN_REQUIRED */
```

Honestly, I am none the wiser after this. I found a little discussion here: https://stackoverflow.com/questions/44318151/why-are-mac-os-x-version-min-required-and-mac-os-x-version-max-allowed-no-longer

**Comment by Albrecht-S** [comment]
@ManoloFLTK, @MatthiasWM : Matthias wrote:

> In any case, the question if we link a library or not should query the same macro that is used to determine if the source code that uses the library is linked in. So should we change the macOSMaxAllowed.c from underscored to non-underscored?

This sounds really wise ;-) and I'd like to add: we should also `#include` the **same header file** (`AvailabilityMacros.h` vs `Availability.h`). I checked it an found that we only `#include <AvailabilityMacros.h>` so this seems to be OK.

Here's another thought: if the `CMake` or `configure` test(s) result in deciding to link the framework (`-weak`) although the **_code_** decides not to use it this will likely not do any harm, but the other way around would obviously be a problem.

Anyway, as long as this question is not solved finally I wouldn't want to release 1.4.0rc3, and please don't expect any further input from me about what's right and what's wrong. I'm asking you, the macOS experts, for the right decision, and I will rely on that.

OTOH, you both checked the **_"yes, I tested it and it's OK"_** checkbox ... should we go ahead with what we have now? Sorry for asking this again, but @MatthiasWM's latest comment seems to open a new case.

_Note: I'll revisit this issue probably after 20:00 CET tonight. If I get your (positive) comments until then I'll try to publish RC3 tonight (or maybe tomorrow, it depends on my spare time tonight)._

**Comment by Albrecht-S** [comment]
OK, thanks, @ManoloFLTK, for the research and detailed info.

So let's go, I'll publish RC3 tomorrow (Thu Nov 7) if nothing else happens.

**Comment by Albrecht-S** [comment]
> I'll publish RC3 tomorrow (Thu Nov 7) if nothing else happens.

Sorry, I was too busy with private stuff to publish the release. Going to work on it this afternoon.

I'm closing this issue now.
